src/constants/MarkdownFiles/posts/gsoc-25-FirePheonix-week01.md

```diff
@@ -0,0 +1,95 @@
+   - By adding a feature to detect color from both uploaded images and a live webcam stream, users would be able to implement Lego music notation for the blind and similarly interactive programs.
+   
+   ![screenshot-description](https://i.ibb.co/d0yVXJmP/Lego-Bricks-Google-Chrome-08-06-2025-01-12-56.png)
+   
+
+2. **Live webcam feed and editing options**  
+   - The following feature helps to use a real time video(webcam) onto the Lego Notation detection interface. Also, you may freely edit and move it around the canvas.
+   ![screenshot-description](https://i.ibb.co/n8cswJsP/Lego-Bricks-Google-Chrome-08-06-2025-01-14-01.png)
+
+   - Here's the reference video regarding lego bricks as musical notes:
+   [youtube: LOfrCPf3XJU]
+
+   - Here's Devin's CMK project for color sensor project in music blocks:
+   [Reflections from constructing modern knowledge](https://medium.com/@sugarlabs/reflections-from-constructing-modern-knowledge-2024-1ce7d60fbb1c) 
+    
+3. **Identified methods to making a new widget block in music blocks**  
+   - I read and went through a lot of documentation, searching for how we can add a new widget block in music blocks. I figured out a few flaws and how can the documentation be improved for future contributors.
+   ![screenshot-description](https://i.ibb.co/bVD8Z54/image.png)
+   - From what I've realized working on it, for adding a new block on the music blocks, I definitely think that for adding a new block, a lot of the code - UIs, features, etc. would already exist and you can just inherit those exisiting classes. And also, you'll have to edit and change a LOT of files and add new methods for your own new block.
+    ![screenshot-description](https://i.ibb.co/hJRKV0Vq/Documentation.png)
+
+---
+
+## Challenges & How I Overcame Them
+
+- **Challenge:** Getting started on features and UIs with a lot of doubts about how it should look like.
+- **Solution:** Consulting my mentors, presenting variations of "How it could be" to them, refering to EXISTING UIs in Music Blocks.
+
+---
+
+## Key Learnings
+
+- Gained familiarity with **making new blocks**.
+- Deepened understanding of **inheritance and code modularity**
+- Improved skills in **exports, imports, code reusability**, **documentation**, and **collaboration workflows**.
+
+---
+
+## Next Week’s Roadmap
+
+- Implement **mapping of musical notes to lego bricks' colors**.
+- Refine **UIs** based on mentor feedback.
+- Inherit the exisiting code inside **WidgetBlocks.js** code as UI.
+- Target: Completing the **core implementation** in the week number 2 and 3.
+
+---
+
+## Acknowledgments
+
+Thanks to some old pull requests and documentation available in music blocks, I was able to figure out on a lot about how new blocks are created on the music blocks interface. Will definitely add on more.
+
+---
+
+## Connect with Me
+
+- GitHub: [@FirePheonix](https://github.com/FirePheonix)
+- Gmail: [shubhsoch@gmail.com](mailto:shubhsoch@gmail.com)
```

Comment:
Do you really want to include your personal email? (You may get more spam...)

Reply:
I see it's in the template. I'm discussing with others, but I think we shouldn't have it...